src/Transports.AspNetCore/GraphQLRequestExecutionResult.cs

```diff
@@ -1,44 +0,0 @@
-using GraphQL.Transport;
-
-namespace GraphQL.Server.Transports.AspNetCore;
-
-/// <summary>
-/// Represents the result of a GraphQL operation. Single GraphQL request may contain several operations, that is, be a batched request.
-/// </summary>
-public readonly struct GraphQLRequestExecutionResult
```

Comment:
Deleting this struct will create some problems for my code. I use elapsed and indexInBatch for logging.

Reply:
I have not yet watched all the code and I suppose that I will have to override some virtual methods to get this information on my own. I will need to inherit from middleware to do so but logging code depends from base middleware and resides in separate package. I don't know yet how destructive these changes for me.

Reply:
I'd have to revisit how this was originally implemented to either see if it makes sense to add back.   Notice I did update one of the samples to catch timing information; however, I'm not entirely sure that it would have the same result as before.  Perhaps you can review some more and determine this.  Ideally I would like to see that overriding the proper method can capture the information you need -- starting a timer and then call `base` and then end timer.  But if for example you specifically want the execution time (excluding response time), this might be a problem.  Then there's batched requests also.  And finally, you could register your own `IDocumentExecuter<>` which captures timing inside there -- but then you don't have the timing for a batched request.

Please consider the options and let me know if you think there is a way for you to capture timing information, or if this is a feature we should consider adding back.

Reply:
>  I did update one of the samples to catch timing information

I saw it. The difference for me is that I catch this info in another package that references **base** API provided by server project.

Regarding other notes from you - I agree. I need to check how my code can be changed. Honestly, I myself do not really like that I added these properties, especially elapsed. At that time it was a simple solution.

Reply:
My use case:
```csharp
 public class GraphQLHttpInterceptionMiddleware<TSchema> : GraphQLHttpMiddleware<TSchema>
        where TSchema : ISchema
    {
        private readonly IExecutionResultInterceptor[] _interceptors;
        private readonly IServiceCallContextAccessor _serviceCallContextAccessor;

        public GraphQLHttpInterceptionMiddleware(
            IGraphQLTextSerializer serializer,
            IOptions<ExecutionOptions> options,
            IEnumerable<IUnhandledExceptionInterceptor> exceptionInterceptors,
            IEnumerable<IExecutionResultInterceptor> executionResultInterceptors,
            IServiceCallContextAccessor serviceCallContextAccessor)
            : base(serializer)
        {
            _interceptors = executionResultInterceptors.ToArray();
            _serviceCallContextAccessor = serviceCallContextAccessor;

            options.Value.UnhandledExceptionDelegate += context =>
            {
                foreach (var handler in exceptionInterceptors)
                    handler.InterceptUnhandledException(context);

                return Task.CompletedTask;
            };
        }

        protected override Task RequestExecutedAsync(in GraphQLRequestExecutionResult requestExecutionResult)
        {
            if (_interceptors.Length > 0)
            {
                var req = new Request(requestExecutionResult.Request.Query, requestExecutionResult.Request.OperationName, requestExecutionResult.Request.Variables);
                foreach (var interceptor in _interceptors)
                    interceptor.InterceptExecutionResult(new RequestExecutionResult(req, requestExecutionResult.Result, requestExecutionResult.Elapsed, requestExecutionResult.IndexInBatch));
            }

            return base.RequestExecutedAsync(requestExecutionResult);
        }

        protected override CancellationToken GetCancellationToken(HttpContext context)
            => _serviceCallContextAccessor.ServiceCallContext?.CancellationToken ?? CancellationToken.None;
    }

  public readonly struct Request : IEquatable<Request>
    {
       
        public Request(string query, string? operationName, Inputs? variables)
        {
            Query = query;
            OperationName = operationName;
            Variables = variables;
        }

      
        public string Query { get; }

     
        public string? OperationName { get; }

   
        public Inputs? Variables { get; }

       
        public override bool Equals(object? obj) => obj is Request other && Equals(other);

     
        public bool Equals(Request other)
            => Query == other.Query &&
               OperationName == other.OperationName &&
               ((Variables == null && other.Variables == null) || (Variables?.Count == other.Variables?.Count && !Variables.Except(other.Variables).Any()));

       
        public override int GetHashCode() => (Query, OperationName, Variables).GetHashCode();

       
        public static bool operator ==(Request left, Request right) => left.Equals(right);

      
        public static bool operator !=(Request left, Request right) => !(left == right);
    }

  public class ExecutionResultLogger : IExecutionResultInterceptor
    {
        private readonly ILogger _logger;
        private readonly ExecutionResultLoggerOptions _options;

    
        public ExecutionResultLogger(ILogger<ExecutionResultLogger> logger, IOptions<ExecutionResultLoggerOptions> options)
        {
            _logger = logger;
            _options = options.Value;
        }

       
        public void InterceptExecutionResult(in RequestExecutionResult requestExecutionResult)
        {
            var level = _options.GetLevel(requestExecutionResult);

            if (_logger.IsEnabled(level))
            {
                var data = new ExecutionResult(requestExecutionResult.Result) { Errors = null };
                var errors = requestExecutionResult.Result.Errors == null ? null : new ExecutionResult(requestExecutionResult.Result) { Data = null, Extensions = null };

                _logger.Log(
                    level,
                    "GraphQL request {GraphQLRequestQuery} completed in {Elapsed} with response {@GraphQLResponse}",
                    requestExecutionResult.Request.Query,
                    (int)requestExecutionResult.Elapsed.TotalMilliseconds,
                    new
                    {
                        Data = Serializer.Serialize(data, _options.Indented).Left(_options.MaxDataLength),
                        Errors = Serializer.Serialize(errors, _options.Indented),
                    });
            }
        }
    }
```

Reply:
>  I use elapsed and indexInBatch for logging.

Only elapsed actually.